api/v1beta1/common_types.go

```diff
@@ -37,6 +37,9 @@ const (
 	// update that disallows a pre-existing Cluster to be populated with Topology information and Class.
 	ClusterTopologyUnsafeUpdateClassNameAnnotation = "unsafe.topology.cluster.x-k8s.io/disable-update-class-name-check"
 
+	// MachinePoolOwnedLabel is the label set on Machines and InfraMachines linked to a MachinePool.
+	MachinePoolOwnedLabel = "machinepool.cluster.x-k8s.io/owned"
```

Comment:
is there any way to reuse `OwnerKindAnnotation` with value `MachinePool` for this instead of adding a new label?

Reply:
I think we'd want to add a label because we can't query for resources using an annotation, i.e. select all Machines owned by a MachinePool. It look like it's a similar situation for the ClusterClass label:
```
	// ClusterTopologyOwnedLabel is the label set on all the object which are managed as part of a ClusterTopology.
	ClusterTopologyOwnedLabel = "topology.cluster.x-k8s.io/owned"
```